packages/interfaces/src/receiver.ts

```diff
@@ -1,17 +1,23 @@
 import type { IDecodedMessage, IDecoder } from "./message.js";
 import type { Callback, ProtocolOptions } from "./protocols.js";
 
-type Unsubscribe = () => void | Promise<void>;
 type PubSubTopic = string;
 type ContentTopic = string;
 
+export type SubscriptionReturn<FilterVersion extends "v1" | "v2"> = {
+  unsubscribe: () => Promise<void>;
+  unsubscribeAll: () => Promise<void>;
+} & (FilterVersion extends "v1" ? object : { ping: () => Promise<void> });
+
 export type ActiveSubscriptions = Map<PubSubTopic, ContentTopic[]>;
 
-export interface IReceiver {
+export interface IReceiver<FilterVersion extends "v1" | "v2"> {
```

Comment:
It's odd because now what does relay implements? `IReceived<v1` ?

I would suggest instead to have an interface that defines the API of Filter v2, and just have the `ping` and other properties being optional.